### PR TITLE
chore: revise input content and order

### PR DESCRIPTION
**Pull request by rossmoody**
This PR introduces updated content from UXCS and revised ordering of examples on the Input component in Forms.

## Checklist

- [ ] :ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)
- [ ] :black_nib: copy updates are approved (add the content strategist as a reviewer)
- [ ] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [ ] :zap: audited via [web.dev](https://web.dev/measure/) for performance and SEO
- [ ] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: src/pages/components/input.mdx [hunk 1]
```diff
@@ -17,17 +17,21 @@ components:
...
   <Use>
     <ul>
-      <li>To let the user enter data</li>
-      <li>To enter multiline text, use a Textarea</li>
+      <li>To let the user enter data into a field</li>
     </ul>
   </Use>
+  <Misuse>
+    <ul>
+      <li>To enter multiline text, use a Textarea</li>
```
Comment (jzempel) [inline src/pages/components/input.mdx:25]: Just a bit odd to see a "Not for this" and, yet, the `Textarea` component exists on this page.

File: src/pages/components/input.mdx [hunk 2]
```diff
@@ -105,8 +120,8 @@ import BareCode from '!!raw-loader!../../examples/forms/inputs/Bare.tsx';
 
 ### Focus inset
 
-Input focus styling can be inset using the `focusInset` property. This keeps the focus contained
-within the input area when it is flush with another component.
+Set focus styling inside the element using the `focusInset` prop. (Useful when an input is flush
```
Comment (jzempel) [inline src/pages/components/input.mdx:123]: Should "input" be capitalized here?

File: src/pages/components/input.mdx (hunk 1 above)
Comment (rossmoody) [reply src/pages/components/input.mdx:25]: Good callout. This seems more like a when to use

Comment (rossmoody) [reply src/pages/components/input.mdx:25]: @vsri1137 You ok with moving this to When to use?

File: src/pages/components/input.mdx (hunk 2 above)
Comment (rossmoody) [reply src/pages/components/input.mdx:123]: @vsri1137 What say ye?

Comment (vsri1137) [reply src/pages/components/input.mdx:123]: Yep.

**Comment by rossmoody** [comment]
@allisonacs This one is ready for your review

**Pull request merged**